src/main/java/org/opentripplanner/apis/transmodel/model/stop/QuayType.java

```diff
@@ -380,36 +359,29 @@ public static GraphQLObjectType create(
           .name("flexibleArea")
           .description("Geometry for flexible area.")
           .type(GeoJSONCoordinatesScalar.getGraphQGeoJSONCoordinatesScalar())
-          .dataFetcher(environment ->
-            (
-              environment.getSource() instanceof AreaStop areaStop
-                ? areaStop.getGeometry().getCoordinates()
-                : null
-            )
-          )
+          .dataFetcher(env -> {
+            StopLocation stopLocation = env.getSource();
+            return stopLocation.getEncompassingAreaGeometry().isEmpty()
+              ? null
+              : stopLocation.getEncompassingAreaGeometry().get().getCoordinates();
```

Comment:
```suggestion
            return stopLocation.getEncompassingAreaGeometry().map(g -> g.getCoordinates()).orElse(null);
```

This is where the Optionals really shine: no more forrest of null checks!

Reply:
Instead a maze of streams.

Reply:
Sorry to be a pedant, but there are actually no streams involved, just monad-like operations on. But I get your point about style preferences.